README.md

```diff
@@ -1,49 +1,207 @@
+[![codecov](https://codecov.io/gh/ballerina-platform/module-ballerinax-redis/branch/master/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerinax-redis)
+[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-redis/actions/workflows/build-with-bal-test-graalvm.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-redis/actions/workflows/build-with-bal-test-graalvm.yml)
 [![GitHub Last Commit](https://img.shields.io/github/last-commit/ballerina-platform/module-ballerinax-redis.svg)](https://github.com/ballerina-platform/module-ballerinax-redis/commits/master)
-[![GraalVM Check](https://github.com/ballerina-platform/module-ballerinax-redis/actions/workflows/build-with-bal-test-native.yml/badge.svg)](https://github.com/ballerina-platform/module-ballerinax-redis/actions/workflows/build-with-bal-test-native.yml)
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
+[![GitHub Issues](https://img.shields.io/github/issues/ballerina-platform/ballerina-library/module/redis.svg?label=Open%20Issues)](https://github.com/ballerina-platform/ballerina-library/labels/module%2Fredis)
+
+[Redis](https://redis.io/) is an open-source, in-memory data structure store that can be used as a database,
+cache, and message broker. It supports various data structures such as strings, hashes, lists, sets, and more.
+
+The `ballerinax/redis` package offers APIs to connect to Redis servers and to manipulate key-value data, providing
+functionality to utilize Redis both as a database and cache. The current connector is compatible Redis server
+versions up to 7.2.x.
+
+## Setup guide
+
+To use the Redis connector, you need to have a Redis server running and accessible. For that, you can either install
+Redis locally or use a cloud-based Redis service.
+
+### Setting up a Redis server locally
+
+#### Step 1: Install Redis server
+1. Download and install Redis from the [official website](https://redis.io/download/).
+
+2. Follow the installation instructions based on your operating system.
+
+#### Step 2: Start Redis server
+After installation, start the Redis server by executing the appropriate command based on your operating system.
+```bash
+redis-server
+```
+
+#### Step 3: Verify Redis connectivity
+Ensure that the Redis server is running and accessible.
+
+```bash
+redis-cli ping
+```
+
+If the server is running, you will receive a response of 'PONG'.
+
+### Setting up a cloud-based Redis service
+
+Several cloud providers offer managed Redis services. Some of the popular ones are:
+
+1. **Redis Labs:** [Redis Labs](https://redis.com/) offers Redis Enterprise Cloud, a fully-managed Redis service in the cloud.
+2. **AWS ElastiCache:** Amazon Web Services provides [ElastiCache](https://aws.amazon.com/elasticache/), a managed Redis service for deploying, operating, and scaling Redis in the AWS Cloud.
```

Comment:
Isn't this Amazon ElastiCache?

Reply:
Yeah that seems to be the more common term. Updated with https://github.com/ballerina-platform/module-ballerinax-redis/pull/220/commits/37c3b4da3f29606fe91def48067c024a7d8f88f7